src/main/java/com/iota/iri/cache/impl/CacheImpl.java

```diff
@@ -0,0 +1,103 @@
+package com.iota.iri.cache.impl;
+
+import com.iota.iri.cache.Cache;
+import com.iota.iri.cache.CacheConfiguration;
+import com.iota.iri.cache.util.ValueWrapper;
+import com.iota.iri.storage.PersistenceProvider;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+
+import com.google.common.collect.MapMaker;
+
+public class CacheImpl implements Cache {
+
+    private final String name;
+    private final CacheConfiguration cacheConfiguration;
+    private final ConcurrentMap<Object, Object> store;
+
+    public CacheImpl(String name) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Public method and constructor comments are required](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4567264)